### DISCUSSION TITLE
FDS does not create a "_devc_ctrl_log.csv" file; smoke detectors do not trigger

### DISCUSSION THREAD

**Discussion by FireResearch-BK** [firemodels/fds#12201] [Q&A] [answered]
I have a case with smoke detectors which record data, but never trigger. There is also no "[Casename]_devc_ctrl_log.csv" file.

[Case.txt](https://github.com/firemodels/fds/files/13186658/Case.txt)

This is for a fire on a particular floor of a building. The exact same setup with fire, smoke detectors and temperature sensors works just fine on another floor of said building.
Geometry obscuration does not appear to be an issue as the smoke detectors record plausible data and eventually do exceed the obscuration threshold (and should therefore trigger) after around 26 seconds (as per _devc.csv). HRR also develops as intended and there is more than enough smoke around. The detectors also show up in Smokeview.

I let the case run for around 250 seconds, but the detectors still do not trigger while logging > 90 % obscuration.

I have compared the .fds files of the working case for the other floor with this one, but can not spot anything amiss.

FDS version is 6.8.0.

As I am unsure if I'm looking at a bug or [PEBKAC](https://en.wiktionary.org/wiki/PEBKAC) error after all, I'm asking here first.

Note: The "out of mesh" devices are semi-intentional/accepted and a product of Pyrosim's handling regarding scenarios and control logic (too complex to explain here), but should not cause any issue. Deactivating or removing them does not improve things.

&nbsp;

- Edit: 

If I simulate the entire building, not just a single floor, I get a "[Casename]_devc_ctrl_log.csv" file and the detectors trigger just fine.

Could it be that FDS does not like meshes that do not start at or near z = 0, at least for devices with a trigger?

**Comment by FireResearch-BK** [comment]
Please reopen this as the original question (as to why devices do not trigger if the mesh does not start at z = 0 m) remains unanswered.

**Reply by drjfloyd** [reply, marked as answer]
After fixing the source code to get the devc log (see issue https://github.com/firemodels/fds/issues/12202), the devc log was showing all four devices changing state between 23 and 27 s.

**Reply by FireResearch-BK** [reply]
I wasn't informed by Github that this was transferred to an actual issue. Thanks for notifying me.
